File: tests/inputs/hello.kt
// Sample Kotlin file to exercise syntax highlighting, indents,
// and folds in vorto. Open with `vorto assets/samples/hello.kt`.

data class Person(
    val name: String,
    val age: Int,
    val tags: List<String> = emptyList(),
) {
    val isAdult: Boolean
        get() = age >= 18
}

enum class Classification(val label: String) {
    NEGATIVE("negative"),
    ZERO("zero"),
    POSITIVE_EVEN("positive even"),
    POSITIVE_ODD("positive odd"),
}

fun greet(person: Person, prefix: String = "Hello"): String =
    "$prefix, ${person.name}!"

fun classify(n: Int): Classification = when {
    n < 0 -> Classification.NEGATIVE
    n == 0 -> Classification.ZERO
    n % 2 == 0 -> Classification.POSITIVE_EVEN
    else -> Classification.POSITIVE_ODD
}

fun squares(numbers: List<Int>): List<Int> =
    numbers.map { it * it }.filter { it > 5 }

fun main() {
    val alice = Person(name = "Alice", age = 30, tags = listOf("admin", "early_bird"))
    println(greet(alice))

    for (n in -2..5) {
        println("$n -> ${classify(n).label}")
    }

    println("squares: ${squares(listOf(1, 2, 3, 4, 5))}")
}

File: tests/inputs/hello.js
// Sample JavaScript file to exercise syntax highlighting, indents,
// and folds in vorto. Open with `vorto assets/samples/hello.js`.

class Person {
  constructor(name, age, tags = []) {
    this.name = name;
    this.age = age;
    this.tags = tags;
  }

  get isAdult() {
    return this.age >= 18;
  }
}

const CLASSES = Object.freeze({
  NEGATIVE: "negative",
  ZERO: "zero",
  POSITIVE_EVEN: "positive even",
  POSITIVE_ODD: "positive odd",
});

function greet(person, prefix = "Hello") {
  return `${prefix}, ${person.name}!`;
}

function classify(n) {
  if (n < 0) return CLASSES.NEGATIVE;
  if (n === 0) return CLASSES.ZERO;
  return n % 2 === 0 ? CLASSES.POSITIVE_EVEN : CLASSES.POSITIVE_ODD;
}

const squares = (numbers) => numbers.map((x) => x * x).filter((x) => x > 5);

function main() {
  const alice = new Person("Alice", 30, ["admin", "early_bird"]);
  console.log(greet(alice));

  for (let n = -2; n <= 5; n++) {
    console.log(`${n} -> ${classify(n)}`);
  }

  console.log("squares:", squares([1, 2, 3, 4, 5]));
}

main();
